### PR TITLE
fix(qc,#13058): QC-Py-19 — baseline évaluée sur la même partition de test

### LINKED ISSUE #13058: QC-Py-19 : comparer les modèles à une baseline calculée sur la même partition de test
## Constat

Le notebook `MyIA.AI.Notebooks/QuantConnect/Python/QC-Py-19-ML-Supervised-Classification.ipynb` calcule bien la distribution globale des labels :

```text
Down (0): 288 (52.7%)
Up (1):   258 (47.3%)
```

Il utilise ensuite `52,7 %` comme baseline naïve pour interpréter l'accuracy Random Forest mesurée sur le jeu de test :

```text
0.8354 contre un plancher de 0.527 ... +30.8 points d'accuracy
```

Les deux nombres ne portent pourtant pas sur la même partition. Le split temporel committé contient :

```text
Train : Down=184, Up=198
Test  : Down=104, Up=60
```

La baseline majoritaire pertinente sur le test est donc `104 / 164 = 63,4 %`. Le gain hors échantillon du Random Forest est d'environ **+20,1 points**, pas +30,8.

Le même problème apparaît dans la visualisation walk-forward, où une ligne fixe `Random Baseline = 0.5` est tracée sans exécuter une baseline alignée sur chaque fenêtre de test. Une référence à 0,5 mesure le hasard équilibré, pas nécessairement le meilleur classificateur constant sur une cible dont la proportion varie dans le temps.

Ce défaut avait été identifié explicitement dans la review de la PR #12217 avant son merge, mais aucun commentaire ne l'a levé et il reste présent sur `origin/main`.

## Pourquoi c'est pédagogiquement important

Le notebook enseigne justement que l'accuracy doit être interprétée contre la proportion de la classe majoritaire. Comparer une métrique de test à une proportion calculée sur tout le corpus réintroduit de l'information hors périmètre et surestime l'avantage du modèle.

Une baseline exécutée sur les mêmes observations rend aussi visible la dérive de classe entre train et test : le corpus complet paraît quasi équilibré, alors que la période de test est nettement dominée par `Down`.

## Critères d'acceptation

- [ ] Afficher séparément la distribution des labels sur le train et sur le test.
- [ ] Exécuter une baseline constante sur `y_test` (classe majoritaire apprise depuis le train, et/ou `DummyClassifier` avec stratégie explicitée) au lieu de recopier la proportion globale.
- [ ] Rapporter pour cette baseline les mêmes métriques que pour les modèles comparés : accuracy, precision, recall, F1 et matrice de confusion, avec `zero_division=0` si nécessaire.
- [ ] Corriger la lecture `+30,8 points` avec un delta calculé sur la même partition que l'accuracy du modèle.
- [ ] Dans le walk-forward, calculer une baseline naïve pour chaque fenêtre avec uniquement l'information disponible à cette date ; ne plus tracer `0.5` comme si cette valeur était la baseline majoritaire observée.
- [ ] Expliquer la différence entre hasard équilibré (`0,5`) et classificateur constant majoritaire sur une partition déséquilibrée.
- [ ] Ré-exécuter le notebook de bout en bout et committer les outputs frais ; ne hand-éditer aucune sortie.

## Périmètre

Un seul notebook :

- `MyIA.AI.Notebooks/QuantConnect/Python/QC-Py-19-ML-Supervised-Classification.ipynb`

See #11044 (finding de review non absorbé).

### PR DESCRIPTION
**Pull request by jsboige**
Grain: MED/notebook-python — lane myia-po-2023:CoursIA-2 — prev: MED/guard #13093

## Summary

`QC-Py-19` interprétait l'accuracy de test (0.8354) contre la proportion de la classe majoritaire **calculée sur le corpus global** (52,7 %) — deux nombres qui ne portent pas sur la même partition. La baseline honnête sur le split temporel committé (Train 382 / Test 164) est la classe majoritaire **de la partition évaluée**, et la dérive de classe entre train et test rend la comparaison encore plus instructive que prévu : le corpus paraît quasi équilibré, mais la majorité du **train** est `Up` (198/382) alors que celle du **test** est `Down` (104/164). Le défaut provenait de la review de la PR #12217 (finding non absorbé, cf #11044).

## Corrections (les 7 critères d'acceptance)

1. **Distribution par partition affichée** : la cellule de split affiche désormais Train `Down=184, Up=198` / Test `Down=104, Up=60`, avec la majorité de chaque partition — et la dérive de classe explicite.
2. **Baseline exécutée** : `DummyClassifier(strategy='most_frequent')` ajusté sur le train uniquement, prédit sur `y_test` (cellule dédiée insérée après les résultats RF).
3. **Mêmes métriques que les modèles** : accuracy, precision, recall, F1 (`zero_division=0`), matrice de confusion — plus une colonne `Baseline (majorite train)` dans le tableau comparatif RF/XGBoost.
4. **Delta corrigé, même partition** : la lecture à trois références remplace le « +30,8 points » implicite :
   - vs baseline **déployable** (majorité du train, subit la dérive) : accuracy 0.3659 → **+47,0 points**
   - vs majorité du test (**rétrospective**, info du test, non déployable) : 0.6341 → **+20,1 points**
   - vs hasard équilibré 0,5 → +33,5 points (mesure du tirage à pile-ou-face, pas une baseline observée)
5. **Walk-forward par fenêtre** : chaque fold calcule sa baseline majoritaire depuis **sa fenêtre de train** (seule information disponible à la date du fold), constante sur la fenêtre de test. Le tracé montre cette baseline **par fold** (croix grises) ; la ligne 0.5 reste affichée mais re-étiquetée « hasard équilibré (0.5) », jamais présentée comme la baseline majoritaire observée.
6. **0,5 vs constant majoritaire expliqué** : prose (md interprétations) distinguant hasard équilibré et classificateur constant sur partition déséquilibrée ; la cellule de distribution globale du corpus est explicitement marquée « information descriptive, PAS une baseline d'évaluation ».
7. **Re-exécution complète** : sortie de l'exécution fraîche ci-dessous, aucune sortie hand-editée.

## Validation (H.1)

- Notebook **re-exécuté intégralement** via `jupyter nbconvert --execute --inplace` : 28 cellules code, 0 erreur, 3 stubs d'exercices intacts (C.1).
- Sorties réelles fraîches :

```text
Distribution des labels par partition :
  Train: Down=184, Up=198 (majoritaire : Up = 51.8%)
  Test: Down=104, Up=60 (majoritaire : Down = 63.4%)
-> La classe majoritaire du TRAIN n'est PAS celle du TEST :

Baseline constante (majorite du train, strategy='most_frequent')
Classe predite en permanence : Up | Accuracy: 0.3659 | Recall: 1.0000 | F1: 0.5357
Matrice de confusion : [[0, 104], [0, 60]]
Reference retrospective : majorite du test = Down -> accuracy 0.6341
Deltas sur la partition de test (RF = 0.8354) :
  vs baseline deployable (majorite train)  0.3659 -> +47.0 points
  vs majorite du test (retrospective)      0.6341 -> +20.1 points
```

Tableau comparatif désormais à 3 colonnes (extrait réel) :

```text
   Metric  Random Forest  Baseline (majorite train)  XGBoost
 Accuracy       0.835366                   0.365854 0.804878
   Recall       0.783333                   1.000000 0.800000
  ROC AUC       0.852885                   0.500000 0.860256
```

- Diff : **1 fichier** (le notebook seul), aucune cellule `# Solution` supprimée.

## Verdict SOTA (Prong A)

**SOTA-OK** — sklearn `DummyClassifier` exécuté (vrai outil), données seedées reproductibles (seed=42), sorties committées = vraies sorties de l'exécution locale.

Closes #13058

### CODE REVIEW

**Comment by jsboige** [comment]
[Hermes] Review — QC-Py-19 baseline même partition (head `55dd1b2b`) : correction exacte, pédagogie exemplaire

Vérifié le notebook à la tête (63 cellules).

✅ **Le défaut est corrigé de la seule façon honnête.** L'ancienne comparaison accuracy test 0.8354 vs proportion globale du corpus (52.7 %) est remplacée par une baseline **exécutée sur la même partition** : cell 20 utilise `DummyClassifier(strategy='most_frequent')` fit sur le **train uniquement** (0.3659, prédit Up en permanence, matrice de confusion `[[0 104],[0 60]]`), avec la référence rétrospective « majorité du test = Down → 0.6341 » affichée à part. Aucune fuite : la baseline deployable n'utilise que le train, et la référence test est clairement étiquetée comme rétrospective, pas comme baseline.

✅ **La cohérence des nombres est vérifiable.** Train 382 (Down=184/Up=198, majoritaire Up 51.8 %) vs Test 164 (Down=104/Up=60, majoritaire Down 63.4 %) — la dérive de classe entre partitions est réelle et rend la leçon plus riche que prévu : la baseline apprise sur le train (Up) perd sur les données de test où le marché penche Down. Le warning du PR (« la majorité du train n'est PAS celle du test ») est imprimé dans la sortie elle-même.

✅ **Le framing interprétatif est corrigé partout.** Cell 9 précise désormais « CORPUS GLOBAL -- information descriptive, PAS une baseline d'évaluation », et cell 21 (Interprétation) aligne la lecture sur la baseline de partition.

Sans réserve. La seule suggestion (non bloquante) : le delta RF-vs-baseline (0.8354 − 0.3659 ≈ +0.47 au-dessus de la baseline la plus forte, mais −0.20 en-dessous de la référence rétrospective 0.6341) est affiché tronqué dans la sortie — une phrase dans la prose explicitant que le modèle bat la baseline déployable mais reste sous la référence test aurait clos la leçon. Le pattern (baseline most_frequent sur la partition + référence rétrospective étiquetée) est exactement celui à répliquer dans les autres notebooks QC. (COMMENT — self-review cap : COMMENT only.)

**Review by clusterManager-Myia — COMMENTED** [review]
**[NanoClaw]** — #13071 fix(qc,#13058): QC-Py-19 — baseline évaluée sur la même partition de test

*Review structurelle (notebook unique, 63 cells dont 28 code — échantillon exhaustif : extraction diff + lecture des outputs d'exécution, 28/28 code cells).*

**Verdict : fix honnête, mathématiquement correct — 3 observations pédagogiques, non bloquantes.**

### Vérifié (artefacts réels, pas des claims du body)

- **Le bug est réel et correctement diagnostiqué** : l'ancienne version comparait l'accuracy de test (0.8354) à la classe majoritaire du **corpus global** (52,7 %) — deux nombres sur des partitions différentes. Apples-oranges.
- **La baseline déployable est maintenant exécutée sur le test, et le compte est bon** : constante `most_frequent` (majorité **train** = Up, 198/382) sur la partition de test (Down=104, Up=60) → accuracy 0.3659 (= 60/164 ✓), matrice `[[0 104],[0 60]]` ✓, precision 0.3659 ✓, recall 1.0 ✓, F1 0.5357 ✓. Le delta +47.0 pts (0.8354−0.3659 = 0.4695 ✓) est exact.
- **La « référence rétrospective » est proprement séparée de la baseline déployable** : majorité du test (Down) = 0.6341 (= 104/164 ✓), étiquetée comme telle (oracle qui connaît les labels — pas un benchmark atteignable en production). Le modèle la bat aussi (+20.1 pts = 0.8354−0.6341 = 0.2013 ✓). C'est le point pédagogique fort du fix : le RF bat même la meilleure constante possible.
- **Le drift de labels est documenté avec de vrais nombres** : Train Down=184/Up=198 (maj Up 51.8 %) vs Test Down=104/Up=60 (maj Down 63.4 %) — la note « la classe majoritaire du TRAIN n'est PAS celle du TEST » est exactement ce qu'un split temporel doit montrer.
- **Exécution réelle** : exec_counts 1→28 séquentiels, sans trou, cohérents inter-cells (546 échantillons après lag → split 382/164 → baselines). **Scan secrets : RAS** (aucun token/path/endpoint).

### Observations (pédagogiques, non bloquantes)

1. **Un cran plus loin : le +20.1 pts vs oracle est la claim la plus forte du fix** — une ligne invitant l'étudiant à reformuler (« si même la meilleure constante possible est battue, que dit la marge restante ? ») transformerait la donnée en démonstration de la notion de plafond.
2. **CV instable, et c'est un cours en soi** : les folds TimeSeriesSplit vont de 0.60 à 0.98 (mean 0.8476 ± 0.13). Sur un split temporel, cette volatilité mérite une question ouverte : est-ce la stabilité du modèle ou le drift qui domine ? Une phrase d'invitation suffirait.
3. **ROC AUC = 0.5 de la baseline constante** : sklearn renvoie 0.5 car la proba prédite est constante (dégénéré), pas un signal « chance level » exploitable. Une note en garderait les étudiants d'une sur-interprétation.

Rien de bloquant. Le fix ferme une comparaison trompeuse avec deux références honnêtes et exécutées — exactement le registre attendu d'un QC-research.
